### PR TITLE
Update to controller-runtime 0.8.1

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
Greetings comrades, this PR updates controller-runtime to the latest version, 0.8.1. This brings us many improvements:

* better context handling in controllers
* better signal handling
* easier health/rediness probes
* stricter fakeclients in tests

Included in this update are

* admissionregistration v1beta1 => v1
* controllers do not hold a context anymore, but instead use the one provide by ctrlruntime when calling the `Reconcile()` function

This PR _does not_ bump the Kubernetes dependencies to 1.20, as this conflicts with the open-policy-agent code. For now we have to live with 1.19.

It's probably a good idea to review commit-by-commit, as it hopefully makes the individual steps much easier to follow. Good luck!

**Does this PR introduce a user-facing change?**:
```release-note
Update to controller-runtime 0.8.1
```

### CODE REVIEW

**Comment by xrstf** [comment]
@zreigz can you please approve instead or in addition to LGTM'ing` :)

/hold
 to give other SIGs time to review

File: cmd/user-cluster-controller-manager/main.go [hunk 1]
```diff
@@ -165,27 +159,22 @@ func main() {
...
 	if err != nil {
 		log.Fatalw("Failed getting user cluster controller config", zap.Error(err))
 	}
-	stopCh := signals.SetupSignalHandler()
-	ctx, ctxDone := context.WithCancel(context.Background())
-	defer ctxDone()
+
+	rootCtx := signals.SetupSignalHandler()
 
 	// Create Context
```
Comment (moelsayed) [inline cmd/user-cluster-controller-manager/main.go]: nit: comment should go, too.

File: pkg/handler/v2/constraint_template/constraint_template_test.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ func TestListConstraintTemplates(t *testing.T) {
 			},
 			HTTPStatus: http.StatusOK,
 			ExistingObjects: test.GenDefaultKubermaticObjects(
+				// test.GenTestSeed(),
```
Comment (moelsayed) [inline pkg/handler/v2/constraint_template/constraint_template_test.go]: is this supposed to be commented or removed?

**Review by moelsayed — COMMENTED** [review]
Two tiny supper nitty comments. Otherwise lgtm.

**Comment by kron4eg** [comment]
This error is E2E config problem, not related to this PR.

/override pre-kubermatic-e2e-openstack-ubuntu-1.20

**Comment by xrstf** [comment]
@moelsayed feedback addressed

**Review by moadqassem — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**